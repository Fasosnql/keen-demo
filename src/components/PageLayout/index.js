import React from 'react';
import { Link } from 'react-router-dom';
import { Grid, Row, Col } from 'react-bootstrap';

import {
  MenuWrapper,
  MenuItem
} from './styles';

const PageLayout = ({ children }) => (
  <Grid>
    <Row>
      <Col xs={12}>
        <MenuWrapper>
          <MenuItem>
            <Link to="/">Home</Link>
          </MenuItem>
          <MenuItem>
            <Link to="/page1">Page1</Link>
          </MenuItem>
          <MenuItem>
            <Link to="/page2">Page2</Link>
          </MenuItem>
          <MenuItem>
            <Link to="/page3">Page3</Link>
          </MenuItem>
          <MenuItem toRight>
            <Link to="/dashboard">Dashboard</Link>
          </MenuItem>
        </MenuWrapper>
      </Col>
    </Row>
    <Row>
      {children}
    </Row>
  </Grid>
);

export default PageLayout;
