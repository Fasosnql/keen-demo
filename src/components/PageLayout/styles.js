import styled, {css} from 'styled-components';

export const MenuWrapper = styled.div`
  background: #eee;
  padding: 15px;
  &:after {
    content: "";
    clear: both;
    display: block;
  }
`;

export const MenuItem = styled.div`
  float: left;
  margin: 0 10px;
  
  a {
    color: #000;
    text-decoration: none;
    
    &:hover {
      color: rgba(0, 0, 0, .8);
    }
  }
  
  ${props => props.toRight && css`
    float: right;
  `}
`;