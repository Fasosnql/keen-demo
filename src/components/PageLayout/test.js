import React from 'react';
import PageLayout from './index';
import { MemoryRouter } from 'react-router-dom';

import snapshotTest from 'utils/snapshot-test.util';

const children = (<div>test children</div>);
snapshotTest('Page Layout component renders correctly', (
  <MemoryRouter>
    <PageLayout children={children} />
  </MemoryRouter>
));