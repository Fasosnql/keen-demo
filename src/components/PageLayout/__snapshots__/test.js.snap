// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Layout component renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-xs-12"
    >
      <div
        className="styles__MenuWrapper-sc-1xqusa2-0 jmRUHa"
      >
        <div
          className="styles__MenuItem-sc-1xqusa2-1 epopFD"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </div>
        <div
          className="styles__MenuItem-sc-1xqusa2-1 epopFD"
        >
          <a
            href="/page1"
            onClick={[Function]}
          >
            Page1
          </a>
        </div>
        <div
          className="styles__MenuItem-sc-1xqusa2-1 epopFD"
        >
          <a
            href="/page2"
            onClick={[Function]}
          >
            Page2
          </a>
        </div>
        <div
          className="styles__MenuItem-sc-1xqusa2-1 epopFD"
        >
          <a
            href="/page3"
            onClick={[Function]}
          >
            Page3
          </a>
        </div>
        <div
          className="styles__MenuItem-sc-1xqusa2-1 etMKix"
        >
          <a
            href="/dashboard"
            onClick={[Function]}
          >
            Dashboard
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div>
      test children
    </div>
  </div>
</div>
`;
